lib/services/peer_info_service.dart

```diff
@@ -22,6 +35,17 @@ class PeerInfoService with ChangeCallbackProvider {
     invokeChangeCallback();
   }
 
+  Future<PeerInfo?> getByID(String id) async {
+    var allDevices = await devices;
+    for (var i = 0; i < allDevices.length; i++) {
+      if (allDevices[i].id == id) {
+        return allDevices[i];
+      }
+    }
+
+    return null;
+  }
+
```

Comment:
Can also be shortened (even though dart streams are a bit weird when finding stuff with null safety and this is actually pretty ugly):
```dart
try {
  return (await devices).firstWhere((device) => device.id == id);
} on StateError {
  return null;
}
```